lib/puppet/functions/type_of.rb

```diff
@@ -0,0 +1,13 @@
+# Returns the type when passed a value.
+#
+# @example how to compare values' types
+#   # compare the types of two values
+#   if type_of($first_value) != type_of($second_value) { fail("first_value and second_value are different types") }
+#
+# See the documentation for "The Puppet Type System" for more information about types.
+#
```

Comment:
Can also add:

Also see the assert_type function for flexible ways to assert the type of a value.